terraform/modules/daily_snapshot/service/list_snapshots.rb

```diff
@@ -46,26 +56,52 @@ def delete
   end
 end
 
-# List the snapshots available in the S3 space hosted by Forest
-def list_snapshots(chain_name = 'calibnet', bucket = 'forest-snapshots', endpoint = 'fra1.digitaloceanspaces.com')
-  ls_format = /\d{4}-\d{2}-\d{2} \d{2}:\d{2}\s*\d*\s*s3:\/\/#{bucket}\/#{chain_name}\/(.+)/
-  snapshot_format = /^([^_]+?)_snapshot_(?<network>[^_]+?)_(?<date>\d{4}-\d{2}-\d{2})_height_(?<height>\d+)(\.forest)?\.car.zst$/
+def prepare_to_list_snapshots(chain_name, bucket)
+  ls_format = %r{\d{4}-\d{2}-\d{2} \d{2}:\d{2}\s*\d*\s*s3://#{bucket}/#{chain_name}/(.+)}
 
   output = `s3cmd ls s3://#{bucket}/#{chain_name}/`
 
+  [ls_format, output]
+end
+
+def prepare_to_update_snapshot_list
+  snapshot_format = \
+    /^(?:[^_]+?)_snapshot_(?<network>[^_]+?)_(?<date>\d{4}-\d{2}-\d{2})_height_(?<height>\d+)(?:\.forest)?\.car.zst$/
   snapshot_list = []
+  [snapshot_format, snapshot_list]
+end
+
+def match_assignments(bucket, chain_name, endpoint, file, match)
```

Comment:
Some documentation for the methods would do, not only this but the other ones (which are doing non-obvious stuff) as well